src/Plutarch/Test/QuickCheck/Instances.hs

```diff
@@ -0,0 +1,974 @@
+    default parbitrary :: (PLift a, Arbitrary (PLifted a)) => Gen (TestableTerm a)
+    parbitrary = pconstantT <$> arbitrary
+
+    pshrink :: TestableTerm a -> [TestableTerm a]
+    pshrink = const []
+
+class PCoArbitrary (a :: S -> Type) where
+    pcoarbitrary :: TestableTerm a -> Gen b -> Gen b
+
+{- | Any Plutarch type that implements `PArbitrary a` automatically gets
+     `Arbitrary` of @TestableTerm a@. This is an interfacing between
+     QuickCheck and Plutarch.
+
+ @since 2.0.0
+-}
+instance PArbitrary p => Arbitrary (TestableTerm p) where
+    arbitrary = parbitrary
+    shrink = pshrink
+
+instance PCoArbitrary p => CoArbitrary (TestableTerm p) where
+    coarbitrary = pcoarbitrary
+
+pdataT :: PIsData p => TestableTerm p -> TestableTerm (PAsData p)
+pdataT (TestableTerm x) = TestableTerm $ pdata x
+
+pfromDataT :: PIsData p => TestableTerm (PAsData p) -> TestableTerm p
+pfromDataT (TestableTerm x) = TestableTerm $ pfromData x
+
+pliftT :: (PLift p, PLifted p ~ h) => TestableTerm p -> h
+pliftT (TestableTerm x) = plift x
+
+pconstantT :: (PLift p, PLifted p ~ h) => h -> TestableTerm p
+pconstantT h = TestableTerm $ pconstant h
+
+pconT :: PlutusType p => (forall s. p s) -> TestableTerm p
+pconT p = TestableTerm $ pcon p
+
+pmatchT :: PlutusType p => TestableTerm p -> (forall s. p s -> Term s b) -> TestableTerm b
+pmatchT (TestableTerm p) f = TestableTerm $ pmatch p f
+
+-- | @since 2.0.0
+instance (PArbitrary p, PIsData p) => PArbitrary (PAsData p) where
+    parbitrary = do
+        (TestableTerm x) <- parbitrary
+        return $ TestableTerm $ pdata x
+    pshrink = fmap pdataT . shrink . pfromDataT
+
+instance (PCoArbitrary p, PIsData p) => PCoArbitrary (PAsData p) where
+    pcoarbitrary (pfromDataT -> x) = pcoarbitrary x
+
+instance Function (TestableTerm PInteger) where
+    function = functionMap pliftT pconstantT
+
+-- | @since 2.0.0
+instance PArbitrary PInteger where
+    pshrink = fmap pconstantT . shrink . pliftT
+
+instance PCoArbitrary PInteger where
+    pcoarbitrary (pliftT -> x) = coarbitrary x
+
+-- | @since 2.0.0
+instance PArbitrary PBool where
+    pshrink = fmap pconstantT . shrink . pliftT
+
+instance PCoArbitrary PBool where
+    pcoarbitrary (pliftT -> x) = coarbitrary x
+
+-- | @since 2.0.0
+instance PArbitrary PUnit where
+    pshrink = fmap pconstantT . shrink . pliftT
+
+instance PCoArbitrary PUnit where
+    pcoarbitrary (pliftT -> x) = coarbitrary x
+
+genByteStringUpto :: Int -> Gen ByteString
+genByteStringUpto m = sized go
+  where
+    go :: Int -> Gen ByteString
+    go s = chooseInt (0, min m s) >>= genByteString
+
+genByteString :: Int -> Gen ByteString
+genByteString l = Exts.fromList <$> vectorOf l arbitrary
+
+shrinkByteString :: ByteString -> [ByteString]
+shrinkByteString bs = do
+    xs' <- shrink . Exts.toList $ bs
+    pure . Exts.fromList $ xs'
+
+-- | @since 2.0.0
+instance PArbitrary PByteString where
+    parbitrary = do
+        len <- chooseInt (0, 64)
```

Comment:
As Vasil lifts this limit, we might want to consider a modifier `PLongByteString` which can produce values over 64 bytes in length.

Reply:
Instead, I think we can just use `sized` in this case.